editor/modes/visual-editor/block.js

```diff
@@ -51,13 +51,28 @@ import {
 	getMultiSelectedBlockUids,
 } from '../../selectors';
 
-const { BACKSPACE, ESCAPE, DELETE, UP, DOWN, LEFT, RIGHT, ENTER } = keycodes;
+const { BACKSPACE, ESCAPE, DELETE, UP, DOWN, LEFT, RIGHT, ENTER, TAB } = keycodes;
+const F10 = 121;
 
 function FirstChild( { children } ) {
 	const childrenArray = Children.toArray( children );
 	return childrenArray[ 0 ] || null;
 }
 
+function selectAncestor( node, selector ) {
+	while ( node !== null ) {
+		if ( node.querySelector( selector ) !== null ) {
+			return node;
+		}
+		node = node.parentNode;
+	}
+	return null;
+}
+
+function isToolbar( target ) {
+	return selectAncestor( target, '.components-toolbar, .editor-block-settings-menu, .editor-block-mover' ) !== null;
```

Comment:
I will read through those links and update my implementation appropriately - as you can probably tell I don't normally work with react.